docs/DeveloperGuide.md

```diff
@@ -2,21 +2,149 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- The command parsing design is inspired by the Command Pattern described in the SE-EDU AddressBook-Level3.
+- Parts of the error-handling and assertion patterns were referenced from Oracle’s Java tutorials.
+- Logger setup referenced from standard Java logging documentation.
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### 1. Nicholas
 
+Responsible for implementing the **Expense** and **Income** management modules. This includes the `Expense`, `Income`, `ExpenseManager`, `IncomeManager`, as well as all associated commands and parsers.
+
+The goal of these modules is to allow users to accurately and efficiently manage their expenses and income with support for adding, editing, deleting, listing, categorizing, and sorting entries.
+
+#### Expense and Income Structure
+
+Each of the two core entities — Expense and Income — is represented by its own class (`Expense`, `Income`) and managed through its own manager class (`ExpenseManager`, `IncomeManager`).
+The below shows the high-level class diagram:
+
+![Expense and Income Structure](img_2.png)
+
+#### Command Parsing and Execution
+
+Each command extends from an abstract `ExpenseCommand` or `IncomeCommand` base class and overrides the `execute(...)` method.
+
+Example sequence diagram for the command `add Lunch 12.5 Food 2025-03-25`:
+
+![Add Expense Command Sequence](img_1.png)
```

Comment:
i think more brevity would make this diagram easier to read, eg removing method arguments (`parseCommand()` instead of `parseCommand(input)`), or just removing the back (dotted) arrows since its pretty straightforward anyways